src/drivers/dw/uart.c

```diff
@@ -0,0 +1,266 @@
+/*
+ * Copyright (c) 2017-2018, Intel Corporation
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *   * Neither the name of the Intel Corporation nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <errno.h>
+#include <stdbool.h>
+#include <string.h>
+
+#include <sof/cpu.h>
+#include <sof/dw-uart.h>
+#include <sof/io.h>
+#include <sof/sof.h>
+#include <sof/string.h>
+#include <sof/wait.h>
+
+#include <platform/platform.h>
+
+#include "uart-priv.h"
+
+struct dw_uart_device_full {
+	struct dw_uart_device common;
+	uint32_t irq;
+	completion_t complete;
+};
+
+static struct dw_uart_device_full uart_dev = {
+	.common = {
+		.port = PLATFORM_LL_UART_REG_BASEADDR,
+		.retry = PLATFORM_LL_UART_RETRY,
+	},
+	/* Keep the CPU field 0 to be able to set it to the CPU, we're running on */
+	.irq = IRQ_EXT_HOST_UART(0),
```

Comment:
This should be a platform macro too.

Reply:
Sorry, could you elaborate a bit. Currently that macro is defined in src/platform/suecreek/include/platform/interrupt.h i.e. it is already platform-specific. Or you mean the whole thing should be
`#define PLATFORM_UART_IRQ IRQ_EXT_HOST_UART(0)`
or similar? Actually wondering what that "EXT_HOST" part of the name is supposed to mean, I think I'll change it too. In fact we can remove that struct from the driver and put it in a platform file, but as long as we don't have any other uses for UARTs as a log console and we don't foresee having more than one of them, we can keep this as is.

Reply:
I mean
```
#define PLATFORM_UART_IRQ IRQ_WHATEVER
```
in platform.h